File: backend/src/main/java/com/coddicted/buzzma/campaign/entity/CampaignStatus.java
package com.coddicted.buzzma.campaign.entity;

public enum CampaignStatus {
  CAMPAIGN_STATUS_DRAFT,
  // Indicates a campaign is closed manually by the owner.
  CAMPAIGN_STATUS_CLOSED,
  // When a Campaign Status is active, campaign details cannot be changed
  // Only Agency can LAUNCH a campaign.
  CAMPAIGN_STATUS_ACTIVE,
  // Campaign Assign - A brand can only assign campaigns to agency
  CAMPAIGN_STATUS_ASSIGNED,
  CAMPAIGN_STATUS_PAUSED,
  CAMPAIGN_STATUS_COMPLETED
}

File: frontend/src/utils/campaignStatus.ts
import type { CampaignStatus } from '../types'

export type RawCampaignStatus =
  | 'CAMPAIGN_STATUS_DRAFT'
  | 'CAMPAIGN_STATUS_CLOSED'
  | 'CAMPAIGN_STATUS_ACTIVE'
  | 'CAMPAIGN_STATUS_ASSIGNED'
  | 'CAMPAIGN_STATUS_PAUSED'
  | 'CAMPAIGN_STATUS_COMPLETED'

export const campaignStatusMap: Record<RawCampaignStatus, CampaignStatus> = {
  CAMPAIGN_STATUS_DRAFT:     'draft',
  CAMPAIGN_STATUS_ACTIVE:    'active',
  CAMPAIGN_STATUS_ASSIGNED:  'active',
  CAMPAIGN_STATUS_PAUSED:    'paused',
  CAMPAIGN_STATUS_COMPLETED: 'completed',
  CAMPAIGN_STATUS_CLOSED:    'completed',
}
